src/controllers/searchResult/base.js

```diff
@@ -97,6 +95,35 @@ class SearchResultBase {
     throw new Error('Unable to retrieve next results from search: missing scrollId, from/sort, or from/size params');
   }
 
+  /**
+   * Automatically fetch each page of results and execute the provided action
+   * on each hit.
+   *
+   * If the action return a promise, this function will wait for all promise to
+   * be resolved.
+   *
+   * @param {Function} action - Action to execute for each hit
+   * @returns {Promise}
+   */
+  async forEachHit(action) {
+    const promises = [];
+    let results = this;
+
+    while (results) {
+      for (const hit of results.hits) {
+        const ret = action(hit);
+
+        if (ret && typeof ret.then === 'function') {
+          promises.push(ret);
+        }
+      }
+
+      results = await results.next();
```

Comment:
Don't we support node 6?

Reply:
Node 6 LTS support ends on the 30/04/2019, but I will use simple promise instead since this is the only `async/await` of the SDK